src/btree/bt_sync.c

```diff
@@ -147,25 +160,26 @@ __sync_ref_list_pop(WT_SESSION_IMPL *session, WT_REF_LIST *rlp, uint32_t flags)
 
     for (i = 0; i < rlp->entry; i++) {
         /*
-         * Mark the obsolete page dirty to let the reconciliation to remove the updates from page.
-         * The obsolete pages with overflow keys cannot be fast deleted, so marking them dirty will
-         * let them clean during reconciliation.
+         * Dirty the obsolete page with overflow items to let the page reconciliation remove all the
+         * overflow items.
          */
-        WT_RET(__wt_page_modify_init(session, rlp->list[i]->page));
-        __wt_page_modify_set(session, rlp->list[i]->page);
+        if (rlp->list[i].ovfl_items) {
+            WT_RET(__wt_page_modify_init(session, rlp->list[i].ref->page));
+            __wt_page_modify_set(session, rlp->list[i].ref->page);
+        }
 
         /*
          * Ignore the failure from urgent eviction. The failed refs are taken care in the next
          * checkpoint.
          */
-        WT_IGNORE_RET_BOOL(__wt_page_evict_urgent(session, rlp->list[i]));
+        WT_IGNORE_RET_BOOL(__wt_page_evict_urgent(session, rlp->list[i].ref));
```

Comment:
Why is this eviction urgent?

Reply:
I don't think it needs an urgent eviction. Other than it clears the cache when the page is removed sooner. And also I feel we don't have any proper exposed API to add a page to the current eviction queue.

Reply:
Makes sense -- all we can really do is set the page generation low enough so that hopefully the page gets picked sooner. My concern here is if we flood the urgent queue with these pages, pages we really might need to urgently evict might get lost in the noise. I'll leave it up to you if you think that's a valid concern -- if we really don't care when these pages make it out, setting the page generation seems sufficient.

Reply:
I agree with your concern. As you suggested, I changed the code to set the page read generation number to a lower value, so that it will be picked for eviction sooner instead of checkpoint cleanup forcing it to evict.